tasks/src/main/scala/dagr/tasks/ScatterGather.scala

```diff
@@ -56,6 +56,16 @@ object ScatterGather {
   /** Provides a factory method to create a [[Scatter]] object from a [[Partitioner]]. */
   object Scatter {
     def apply[Result](partitioner: Partitioner[Result]): Scatter[Result] = partitioner.scatter
+
+    /** Builds a [[Scatter]] from a method that returns a sequence of results. */
+    def apply[Result](toResults: => Seq[Result]): Scatter[Result] = {
```

Comment:
One last thought .. I think if you want to merge this with minimal changes, I would just change the argument type to be `Seq[Result]` and change the docs to say that it creates a scatter from a set of pre-generated partitions and call it a day.